### PR TITLE
Server: Fixes #9931: Add task to delete events older than a week

### LINKED ISSUE #9931: Server - Table events grow constantly
### Operating system

Linux

### Joplin version

Server - v2.13.3

### Desktop version info

Server Joplin  v2.13.3 installed on docker with PostgreSQL 

### Current behaviour

Table events has milion of records, and grows every second.
image
here's an example of the latest line on my DB, as a workaround can oldest record can be deleted from this table or it will brake something on server sync?
I have 15MB of notes (attachments included) and 400MB of events table
image

### Expected behaviour

Events table should be deleted by a rolling windows

### Logs

_No response_

### PR DESCRIPTION
**Pull request by tecc**
This fixes #9931 through adding a task that deletes any tasks older than a week.

The task in question looks like this:
```ts
public async deleteOldEvents(beforeMillis: number = Date.now() - Week) {
	return this.withTransaction(async () => {
		await this.db(this.tableName)
			.where('created_time', '<', beforeMillis)
			.delete();
	}, 'EventModel::deleteOldEvents');
}
```

### CODE REVIEW

**Comment by tecc** [comment]
I have read the CLA Document and I hereby sign the CLA

File: packages/server/src/models/EventModel.ts [hunk 1]
```diff
@@ -33,4 +34,11 @@ export default class EventModel extends BaseModel<Event> {
 			.first();
 	}
 
+	public async deleteOldEvents(beforeMillis: number = Date.now() - Week) {
```
Comment (laurent22) [inline packages/server/src/models/EventModel.ts]: All time should be milliseconds so there's no need to append a suffix. It's only in the rare case when it's not milliseconds that we append one.

No need for the weird default argument value, please make it required.

**Review by laurent22 — COMMENTED** [review]
If something is batch deleting data, surely we need some tests too.

**Comment by tecc** [comment]
Added a test for it, although as a side effect I felt it necessary to make `created_time` a parameter.
Making `created_time` a parameter makes it easier to control the timestamps of the events rather than haphazardly guessing them. The parameter could be used in the `should get the latest event` test as well to speed it up ever so slightly.

File: packages/server/src/models/EventModel.ts [hunk 2]
```diff
@@ -16,11 +15,11 @@ export default class EventModel extends BaseModel<Event> {
 		return UuidType.Native;
 	}
 
-	public async create(type: EventType, name = '') {
+	public async create(type: EventType, name = '', created_time = Date.now()) {
```
Comment (laurent22) [inline packages/server/src/models/EventModel.ts:18]: This is not necessary and should be avoided. Instead you can use this in the test to simulate time:

jest
  .useFakeTimers()
  .setSystemTime(new Date('2020-01-01'));

See here for more information: https://stackoverflow.com/a/63377110/561309

**Comment by laurent22** [comment]
@tecc, sorry I need to close this pull request since we're trying to keep the list of PR current. However if you would like to complete the pull request please let me know and I'll reopen.

**Comment by rainforest1155** [comment]
@tecc Thank you for your work on this so far! I was wondering if you think you would be able to work on this again in the near future? I have about 210MB of data on my Joplin server right now, but the database dump amounts to almost 1000MB, which with my daily backup strategy does add up quite a bit. So I would love to have a clean up task in place as you proposed.
Thanks!

**Pull request closed without merging**